robot-mode.el

```diff
@@ -194,6 +201,10 @@ Used as `indent-line-function' of the mode."
 	    (string-width (buffer-substring-no-properties
 			   (line-beginning-position) (point)))))
 
+	 ;; Current point in line as width
+	 (point-in-line (string-width (buffer-substring-no-properties
+			   (line-beginning-position) (point))))
+
 	 ;; The non-indented contents of the current line
 	 current-line
 
```

Comment:
Maybe set the current-line at this stage and not in the `current-indent` to be more clear.
Also note that there is a built-in function `(current-indentation)`, you could just use it instead of calculating it with `string-width`.